### DISCUSSION TITLE
Is it possible to rotate a previously signed pdf?

### DISCUSSION THREAD

**Discussion by dennisbottaro** [MatthiasValvekens/pyHanko#419] [General]
Hey gang,

I have need on a semi-regular basis to rotate previously signed PDF files.  My use case, our house plans are sent to an outside engineer for approval and signature - sometimes they send us back the signed pdf file with the wrong page orientation.  When we go to submit plans for review all pages must be in correct orientation (yes, it seems it is too much to ask for a user to just rotate the document during review), and we get the submission kicked back when pages aren't properly rotated.  This is adding up to weeks of delays across many jobs and I would love to be able to rotate the signed pdf files prior to sending for plan review.

Is there a way to rotate and preserve the validity of existing signatures?

Thanks for you help!

**Comment by MatthiasValvekens** [comment]
In one word: no :). Or at least not with the way PDF signatures are used today.

You could try fixing the orientation in an incremental update, but I'm pretty sure that most validators that actually check this stuff would reject the signature anyway, since it requires messing the coordinates of the signature appearance as well.

**Comment by dennisbottaro** [comment]
That's along the lines of what I was thinking.   I do appreciate your response!  Thank you!
